youtube_dl/extractor/youtube.py

```diff
@@ -1502,8 +1571,51 @@ def add_dash_mpd(video_info):
             if dash_mpd and dash_mpd[0] not in dash_mpds:
                 dash_mpds.append(dash_mpd[0])
 
+        def get_storyboards(video_info):
+            storyboards = []
+            spec = video_info.get('storyboard_spec', [])
+
+            for s in spec:
+                s_parts = s.split('|')
+                base_url = s_parts[0]
+                i = 0
+                for params in s_parts[1:]:
+                    storyboard_attrib = params.split('#')
+                    if len(storyboard_attrib) != 8:
+                        self._downloader.report_warning('Unable to extract storyboard')
+                        continue
+
+                    frame_width = int_or_none(storyboard_attrib[0])
+                    frame_height = int_or_none(storyboard_attrib[1])
+                    frames = int_or_none(storyboard_attrib[2])
+                    cols = int_or_none(storyboard_attrib[3])
+                    rows = int_or_none(storyboard_attrib[4])
+                    filename = storyboard_attrib[6]
+                    sigh = storyboard_attrib[7]
+
+                    if frame_width and frame_height and cols and rows and frames:
+                        width, height = frame_width * cols, frame_height * rows
+                        n_images = int(math.ceil(frames / float(cols * rows)))
+                    else:
+                        self._downloader.report_warning('Unable to extract storyboard')
+                        continue
+
+                    storyboards_url = base_url.replace('$L', compat_str(i)) + '?'
+                    for j in range(n_images):
+                        url = storyboards_url.replace('$N', filename).replace('$M', compat_str(j)) + 'sigh=' + sigh
+                        storyboards.append({
+                            'id': 'L' + compat_str(i) + '-M' + compat_str(j),
+                            'width': width,
+                            'height': height,
+                            'url': url
```

Comment:
Could you also add 'frames', and maybe 'frame_width' and 'frame_height' or 'cols' and 'rows'?